nav2_controller/src/nav2_controller.cpp

```diff
@@ -52,24 +56,38 @@ ControllerServer::~ControllerServer()
 nav2_util::CallbackReturn
 ControllerServer::on_configure(const rclcpp_lifecycle::State & state)
 {
-  std::string local_planner_name;
-
-  RCLCPP_INFO(get_logger(), "Configuring local controller interface");
+  RCLCPP_INFO(get_logger(), "Configuring controller interface");
 
   costmap_ros_->on_configure(state);
 
   auto node = shared_from_this();
 
   progress_checker_ = std::make_unique<ProgressChecker>(rclcpp_node_);
 
-  try {
-    std::string local_controller_plugin;
-    get_parameter("local_controller_plugin", local_controller_plugin);
-    local_planner_ = lp_loader_.createUniqueInstance(local_controller_plugin);
-    RCLCPP_INFO(get_logger(), "Created local controller : %s", local_controller_plugin.c_str());
-    local_planner_->configure(node, costmap_ros_->getTfBuffer(), costmap_ros_);
-  } catch (const pluginlib::PluginlibException & ex) {
-    RCLCPP_FATAL(get_logger(), "Failed to create local planner. Exception: %s", ex.what());
+  if (controller_types_.size() != controller_names_.size()) {
+    RCLCPP_FATAL(get_logger(), "Size of controller names (%i) and "
+      "controller types (%i) are not the same!",
+      static_cast<int>(controller_types_.size()),
+      static_cast<int>(controller_names_.size()));
+    exit(-1);
+  }
+
+  for (uint i = 0; i != controller_names_.size(); i++) {
+    try {
+      nav2_core::LocalPlanner::Ptr controller =
+        lp_loader_.createUniqueInstance(controller_types_[i]);
+      RCLCPP_INFO(get_logger(), "Created controller : %s of type %s",
+        controller_names_[i].c_str(), controller_types_[i].c_str());
+      controller->configure(node, controller_names_[i],
+        costmap_ros_->getTfBuffer(), costmap_ros_);
+      controllers_.insert({controller_names_[i], controller});
```

Comment:
I find it more clear if this was `controllers_.insert({motion_type[i], controller});`

Reply:
The phrase `motion_type` I think is ambiguous and isn't immediately clear to me what that is. I can see that this has caused you some confusion, would adding `plugin` to these variable names help? `controller_plugin_name` `controller_plugin_type`.